### PR TITLE
Added SubscribeStar provider 

### PR DESCRIPTION
**Pull request by Jnewbon**
Adde OAuth2 provider for [SubscribeStar](https://www.subscribestar.com/)

Docs: https://www.subscribestar.com/api

### CODE REVIEW

File: src/SubscribeStar/Provider.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            'query' => $query,
+        ];
+        $response = $this->getHttpClient()->post(
+            'https://www.subscribestar.com/api/graphql/v1',
+            [
+                'headers' => [
+                    'Accept'        => 'application/json',
+                    'Authorization' => 'Bearer '.$token,
+                ],
+                RequestOptions::FORM_PARAMS => $queryParams,
```
Comment (atymic) [inline src/SubscribeStar/Provider.php:67]:
```suggestion
                RequestOptions::FORM_PARAMS => ['query' => $query],
```

File: src/SubscribeStar/Provider.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            user {
+                id
+                email
+                name
+                avatar_url
+            }
+        }
+        ';
+
+        $queryParams = [
```
Comment (atymic) [inline src/SubscribeStar/Provider.php:57]: Remove this

File: src/SubscribeStar/Provider.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace SocialiteProviders\Subscribestar;
+
+use GuzzleHttp\RequestOptions;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
```
Comment (atymic) [inline src/SubscribeStar/Provider.php:9]: Can we add an `@see` to https://www.subscribestar.com/api in the docblock

File: src/SubscribeStar/Provider.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id'       => $user['data']['user']['id'],
```
Comment (atymic) [inline src/SubscribeStar/Provider.php:80]: Maybe add a null coalesce for email/name is case scopes are overridden?

File: src/SubscribeStar/composer.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            "email": "github@jamienewbon.co.uk"
+        }
+    ],
+    "support": {
+        "issues": "https://github.com/socialiteproviders/providers/issues",
+        "source": "https://github.com/socialiteproviders/providers",
+        "docs": "https://socialiteproviders.com/subscribestar"
+    },
+    "require": {
+        "php": "^7.4 || ^8.0",
```
Comment (atymic) [inline src/SubscribeStar/composer.json:24]:
```suggestion
        "php": "^8.0",
```

File: src/SubscribeStar/composer.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        "ext-json": "*",
+        "socialiteproviders/manager": "~4.0"
```
Comment (atymic) [inline src/SubscribeStar/composer.json:26]:
```suggestion
        "socialiteproviders/manager": "~4.2"
```

File: src/SubscribeStar/Provider.php [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+            'name'     => $user['data']['user']['name'],
+            'email'    => $user['data']['user']['email'],
+            'avatar'   => $user['data']['user']['avatar_url'],
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenFields($code)
```
Comment (atymic) [inline src/SubscribeStar/Provider.php:90]: Can remove this block with bump of the manager to 4.2

**Review by atymic — CHANGES_REQUESTED** [review]

File: monorepo-builder.yml [hunk 8]
```diff
@@ -166,6 +166,7 @@ parameters:
     src/StreamElements: 'git@github.com:SocialiteProviders/StreamElements.git'
     src/Streamlabs: 'git@github.com:SocialiteProviders/Streamlabs.git'
     src/Stripe: 'git@github.com:SocialiteProviders/Stripe.git'
+    src/SubscribeStar: 'git@github.com:SocialiteProviders/SubscribeStar.git'
```
Comment (lucasmichot) [inline monorepo-builder.yml:169]:
```suggestion
    src/Subscribestar: 'git@github.com:SocialiteProviders/Subscribestar.git'
```

File: src/SubscribeStar/Provider.php [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+            [
+                'headers' => [
+                    'Accept'        => 'application/json',
+                    'Authorization' => 'Bearer '.$token,
+                ],
+                RequestOptions::FORM_PARAMS => $queryParams,
+            ]
+        );
+
+        return json_decode($response->getBody(), true);
```
Comment (lucasmichot) [inline src/SubscribeStar/Provider.php:71]:
```suggestion
        return json_decode((string) $response->getBody(), true);
```

File: src/SubscribeStar/README.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
+# Patreon
```
Comment (lucasmichot) [inline src/SubscribeStar/README.md:1]:
```suggestion
# Subscribestar
```

**Review by lucasmichot — CHANGES_REQUESTED** [review]

File: src/SubscribeStar/README.md (hunk 10 above)
Comment (Jnewbon) [reply src/SubscribeStar/README.md:1]: oof How'd I miss that one search and replace failed meee :)

**Comment by Jnewbon** [comment]
With SubscribeStar being a subscription API the users of this might want to grab more data then just the user account. So i was thinking of moving the following

https://github.com/SocialiteProviders/Providers/blob/fc151c8988753b02c86678fd5146c9a628b82234/src/SubscribeStar/Provider.php#L50-L58

into the parameters variable for example (with the extra info)
```php
    protected $parameters = [
        'query' => '{
        user {
            id
            email
            name
            avatar_url
        }
        subscriber {
            subscription {
                id
                active
                billing_failed
                cancelled
                price
            }
        }
    }'
    ];
```
and just have that variable as the form params
`RequestOptions::FORM_PARAMS => $this->parameters`
Think that a valid use of the parameters variable?

**Comment by Jnewbon** [comment]
Wow, my re request review button don't like you two...

File: src/SubscribeStar/Provider.php [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id'       => $user['data']['user']['id'],
+            'name'     => $user['data']['user']['name'] ??= '',
```
Comment (atymic) [inline src/SubscribeStar/Provider.php:81]: Can we default to null not empty string please

**Comment by atymic** [comment]
Maybe we can allow configuring the graphql query via config? Or just leave it so the user can make another query to get more info once they have the access token

File: src/SubscribeStar/Provider.php [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+                name
+                avatar_url
+            }
+        }
+        ';
+
+        $response = $this->getHttpClient()->post(
+            'https://www.subscribestar.com/api/graphql/v1',
+            [
+                'headers' => [
```
Comment (lucasmichot) [inline src/SubscribeStar/Provider.php:63]:
```suggestion
                RequestOptions::HEADERS => [
```

File: src/SubscribeStar/Provider.php [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id'       => $user['data']['user']['id'],
+            'name'     => $user['data']['user']['name'] ??= null,
```
Comment (lucasmichot) [inline src/SubscribeStar/Provider.php:81]:
```suggestion
            'name'     => $user['data']['user']['name'] ?? null,
```

File: src/SubscribeStar/Provider.php [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+            'email'    => $user['data']['user']['email'] ??= null,
```
Comment (lucasmichot) [inline src/SubscribeStar/Provider.php:82]:
```suggestion
            'email'    => $user['data']['user']['email'] ?? null,
```

File: src/SubscribeStar/Provider.php [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+            'avatar'   => $user['data']['user']['avatar_url'] ??= null,
```
Comment (lucasmichot) [inline src/SubscribeStar/Provider.php:83]:
```suggestion
            'avatar'   => $user['data']['user']['avatar_url'] ?? null,
```

File: src/SubscribeStar/composer.json [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+    ],
+    "support": {
+        "issues": "https://github.com/socialiteproviders/providers/issues",
+        "source": "https://github.com/socialiteproviders/providers",
+        "docs": "https://socialiteproviders.com/subscribestar"
+    },
+    "require": {
+        "php": "^8.0",
+        "ext-json": "*",
+        "socialiteproviders/manager": "~4.2"
```
Comment (lucasmichot) [inline src/SubscribeStar/composer.json:26]:
```suggestion
        "socialiteproviders/manager": "^4.2"
```

**Review by lucasmichot — DISMISSED** [review]

**Comment by atymic** [comment]
Please pr to website https://github.com/SocialiteProviders/website

**Review by atymic — APPROVED** [review]

**Pull request merged**